extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQWorkerTaskLauncher.java

```diff
@@ -345,6 +345,26 @@ private void runNewTasks()
     }
   }
 
+  /**
+   * Returns the number of worker tasks that are not yet fully started.
+   */
+  public int getPendingWorkerTasks()
```

Comment:
A small question regarding synchronization, does having separate calls for pending worker tasks and running worker tasks can give the wrong results? (i.e. `fullyStarteedTasks() + pendingWorkerTasks() != desiredTaskCount).
If so, then I wonder if it would be appropriate to have a single call in the ControllerImpl to get the pending (or running tasks) and calculate the counterpart by subtracting it from the desiredTaskCount.

Reply:
Good catch, I have changed it to a single call instead.